kong/conf_loader/init.lua

```diff
@@ -73,244 +72,25 @@ ffi.cdef([[
-    ignore = EMPTY,
-  },
-}
-
-
-local DEPRECATED_DYNAMIC_KEY_NAMESPACES = {}
-
-
-local PREFIX_PATHS = {
-  nginx_pid = {"pids", "nginx.pid"},
-  nginx_err_logs = {"logs", "error.log"},
-  nginx_acc_logs = {"logs", "access.log"},
-  admin_acc_logs = {"logs", "admin_access.log"},
-  nginx_conf = {"nginx.conf"},
-  nginx_kong_gui_include_conf = {"nginx-kong-gui-include.conf"},
-  nginx_kong_conf = {"nginx-kong.conf"},
-  nginx_kong_stream_conf = {"nginx-kong-stream.conf"},
-  nginx_inject_conf = {"nginx-inject.conf"},
-  nginx_kong_inject_conf = {"nginx-kong-inject.conf"},
-  nginx_kong_stream_inject_conf = {"nginx-kong-stream-inject.conf"},
-
-  kong_env = {".kong_env"},
-  kong_process_secrets = {".kong_process_secrets"},
-
-  ssl_cert_csr_default = {"ssl", "kong-default.csr"},
-  ssl_cert_default = {"ssl", "kong-default.crt"},
-  ssl_cert_key_default = {"ssl", "kong-default.key"},
-  ssl_cert_default_ecdsa = {"ssl", "kong-default-ecdsa.crt"},
-  ssl_cert_key_default_ecdsa = {"ssl", "kong-default-ecdsa.key"},
-
-  client_ssl_cert_default = {"ssl", "kong-default.crt"},
-  client_ssl_cert_key_default = {"ssl", "kong-default.key"},
-
-  admin_ssl_cert_default = {"ssl", "admin-kong-default.crt"},
-  admin_ssl_cert_key_default = {"ssl", "admin-kong-default.key"},
-  admin_ssl_cert_default_ecdsa = {"ssl", "admin-kong-default-ecdsa.crt"},
-  admin_ssl_cert_key_default_ecdsa = {"ssl", "admin-kong-default-ecdsa.key"},
-
-  admin_gui_ssl_cert_default = {"ssl", "admin-gui-kong-default.crt"},
-  admin_gui_ssl_cert_key_default = {"ssl", "admin-gui-kong-default.key"},
-  admin_gui_ssl_cert_default_ecdsa = {"ssl", "admin-gui-kong-default-ecdsa.crt"},
-  admin_gui_ssl_cert_key_default_ecdsa = {"ssl", "admin-gui-kong-default-ecdsa.key"},
-
-  status_ssl_cert_default = {"ssl", "status-kong-default.crt"},
-  status_ssl_cert_key_default = {"ssl", "status-kong-default.key"},
-  status_ssl_cert_default_ecdsa = {"ssl", "status-kong-default-ecdsa.crt"},
-  status_ssl_cert_key_default_ecdsa = {"ssl", "status-kong-default-ecdsa.key"},
-}
+local CIPHER_SUITES = conf_constants.CIPHER_SUITES
+local DEFAULT_PATHS = conf_constants.DEFAULT_PATHS
+local HEADER_KEY_TO_NAME = conf_constants.HEADER_KEY_TO_NAME
+local UPSTREAM_HEADER_KEY_TO_NAME = conf_constants.UPSTREAM_HEADER_KEY_TO_NAME
+local DYNAMIC_KEY_NAMESPACES = conf_constants.DYNAMIC_KEY_NAMESPACES
+local DEPRECATED_DYNAMIC_KEY_NAMESPACES = conf_constants.DEPRECATED_DYNAMIC_KEY_NAMESPACES
+local PREFIX_PATHS = conf_constants.PREFIX_PATHS
+local CONF_PARSERS = conf_constants.CONF_PARSERS
+local CONF_SENSITIVE_PLACEHOLDER = conf_constants.CONF_SENSITIVE_PLACEHOLDER
+local CONF_SENSITIVE = conf_constants.CONF_SENSITIVE
+local CONF_BASIC = conf_constants.CONF_BASIC
+local TYP_CHECKS = conf_constants.TYP_CHECKS
+local HEADERS = conf_constants.HEADERS
+local BUNDLED_VAULTS = conf_constants.BUNDLED_VAULTS
+local BUNDLED_PLUGINS = conf_constants.BUNDLED_PLUGINS
+local LMDB_VALIDATION_TAG = conf_constants.LMDB_VALIDATION_TAG
+
```

Comment:
I would actually suggest to not cache those variables, as they are only used once and will cause an extra redirect
when reading the code.

Reply:
Some are used once (`TYP_CHECKS` CIPHER_SUITES), but others are not, should we only localize the latter?